docs/source/en/add_tensorflow_model.mdx

```diff
@@ -0,0 +1,342 @@
+architecture ([example](https://github.com/huggingface/transformers/tree/main/src/transformers/models/bert)).
+
+</Tip>
+
+
+## Step-by-step guide to add TensorFlow model architecture code
+
+There are many ways to design a large model architecture, and multiple ways of implementing said design. However,
+you might recall from our [general overview of 🤗 Transformers](add_new_model#general-overview-of-transformers)
+that we are an opinionated bunch - the ease of use of 🤗 Transformers relies on consistent design choices. From
+experience, we can tell you a few important things about adding TensorFlow models:
+
+- Don't reinvent the wheel! More often that not, there are at least two reference implementations you should check: the
+PyTorch equivalent of the model you are implementing and other TensorFlow models for the same class of problems.
+- Great model implementations survive the test of time. This doesn't happen because the code is pretty, but rather
+because the code is clear, easy to debug and build upon. If you make the life of the maintainers easy with your
+TensorFlow implementation, by replicating the same patterns as in other TensorFlow models and minimizing the mismatch
+to the PyTorch implementation, you ensure your contribution will be long lived.
+- Ask for help when you're stuck! The 🤗 Transformers is here to help, and we've probably found solutions to the same
+problems you're facing.
+
+Here's an overview of the steps needed to add a TensorFlow model architecture:
+1. Select the model you wish to convert
+2. Prepare transformers dev environment
+3. (Optional) Understand theoretical aspects and the existing implementation
+4. Implement the model architecture
+5. Implement model tests
+6. Submit the pull request
+7. (Optional) Build demos and share with the world
+
+### 1.-3. Prepare your model contribution
+
+**1. Select the model you wish to convert**
+
+Let's start off with the basics: the first thing you need to know is the architecture you want to convert. If you
+haven't your eyes set on a specific architecture, asking the 🤗 Transformers team for suggestions is a great way to
+maximize your impact - we will guide you towards the most prominent architectures that are missing on the TensorFlow
+side. If the specific model you want to use with TensorFlow already has a TensorFlow architecture implementation in
+🤗 Transformers but is lacking weights, feel free to jump straight into the weight conversion section of this page.
+
+For simplicity, the remainder of this guide assumes you've decided to contribute with the TensorFlow version of
+*BrandNewBert* (the same example as in the [guide](add_new_model) to add a new model from scratch).
+
+<Tip>
+
+Before starting the work on a TensorFlow model architecture, double-check that there is no ongoing effort to do so.
+You can search for `TensorFlow BrandNewBert` on the [pull request GitHub page](https://github.com/huggingface/transformers/pulls).
```

Comment:
Might also be just `TF BrandNewBert` sometime people are lazy ;-)

Reply:
added a more general tip (search for `BrandNewBert` is enough to find a potential TF PR)